### PR TITLE
NullHtmlStringLocalizer should use NullStringLocalizer

### PR DESCRIPTION
**Pull request by hishamco**

### CODE REVIEW

File: src/OrchardCore/OrchardCore/Localization/NullHtmlLocalizerFactory.cs [hunk 1]
```diff
@@ -43,17 +42,26 @@ public IEnumerable<LocalizedString> GetAllStrings(bool includeParentCultures)
...
+                => StringLocalizer.GetString(name, arguments);
 
-            IHtmlLocalizer IHtmlLocalizer.WithCulture(CultureInfo culture) => Instance;
+            IHtmlLocalizer IHtmlLocalizer.WithCulture(CultureInfo culture)
+                => new NullHtmlLocalizer();
+
+            private static IStringLocalizer StringLocalizer
+                => NullStringLocalizerFactory.NullStringLocalizer.Instance;
+
+            private LocalizedHtmlString ToHtmlString(LocalizedString localizedString)
```
Comment (sebastienros) [inline src/OrchardCore/OrchardCore/Localization/NullHtmlLocalizerFactory.cs]: This is the only change I don't like. This method has no added value, it just makes the code more complex.

Comment (hishamco) [reply src/OrchardCore/OrchardCore/Localization/NullHtmlLocalizerFactory.cs]: It just uses the `LocalizedString` behind the scene, I saw similar approach in `HtmlStringLocalizer`, if you don't like it we can use `LocalizedString` instance directly in the return

**Comment by Skrypt** [comment]
To be honest, I need to understand what this will fix. I don't have any repro steps to know what bug fix I'm looking at. I'm leaving this one to @sebastienros . I'm too much out of context to review this.

**Comment by sebastienros** [comment]
What's left in this PR is only putting the false impression that it's reusing some code, but technically it's just another direct reference to NullStringLicalizer, which as it's name states, does nothing. So the code before was just doing it, nothing. Now it takes a PhD to understand that it does nothing. In the end the only thing to keep in this PR is that the names if the classes have changed. 

Either a PR is self explanatory, or at least it should state what problem it's trying to solve.

**Comment by hishamco** [comment]
> To be honest, I need to understand what this will fix. I don't have any repro steps to know what bug fix I'm looking at.

There's no direct bug here to fix, but this will simplify the things, so in the future if there's a localization bug, I should test `NullStringLocalizer` only coz `NullHtmlLocalizer` will depend heavily on it 

> Now it takes a PhD to understand that it does nothing

:smile:

>  In the end the only thing to keep in this PR is that the names if the classes have changed

Sure

**Comment by sebastienros** [comment]
I think we don't live in the same planet. You now removed the only thing that was acceptable. I am closing this PR becausevwe spend too much time going in circles over literally nothing valuable.

**Comment by hishamco** [comment]
> I think we don't live in the same planet. You now removed the only thing that was acceptable

😞

> In the end the only thing to keep in this PR is that the names if the classes have changed

 I think I miss understood your words> I think we don't live in the same planet. You now removed the only thing that was acceptable

😞

> In the end the **only thing to keep in this PR is that the names** if the classes have changed

May be I miss understood your words

**Comment by sebastienros** [comment]
Exactly: "only thing to keep in this PR is that the names have changed" then you proceed by reverting the name changes.

**Comment by hishamco** [comment]
Nah, I miss understand you .. sorry

**Comment by hishamco** [comment]
If it's only the names needs to change I don't see any value at all coz the class names will not change, also they are private, my intend to reduce unexpected bugs in the future coz the main null localizer will be `NullStringLocalizer`

**Pull request closed without merging**